main.go

```diff
@@ -140,31 +138,32 @@ func main() {
 			cfg.BlockRangeLimit = blockRangeLimit
 		}
 
-		return cfg.run(lgr)
+		return cfg.run(ctx)
 	}
 
+	ctx = gotils.With(ctx, "runtime", time.Since(start))
```

Comment:
It looks like this isn't measuring the actual run time anymore, but also that seems unimportant to log about so it could probably just be dropped.
```suggestion
```